sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewViewModel.groovy

```diff
@@ -14,23 +15,25 @@ import life.qbic.datamodel.samples.Sample
 */
 class ProjectOverviewViewModel {
 
-    List<Project> projects = []
+    ObservableList projects = new ArrayList<Project>()
+    private final ResourceService<Project> projectResourceService
 
-    ProjectOverviewViewModel(){
-        generateMockData()
-    }
-
-    void generateMockData(){
-        Location location = new Location()
+    Subscriber<Project> projectAddedSubscription
+    Subscriber<Project> projectRemovedSubscription
 
-        Sample sample = new Sample()
-        sample.setCode("Q1234AC")
-        sample.setCurrentLocation(location)
+    ProjectOverviewViewModel(ResourceService<Project> projectResourceService){
+        this.projectResourceService = projectResourceService
+        fetchProjectData()
+        subscribeToResources()
+    }
 
-        Project project1 = new Project.Builder("Q1234","This is a test project",[sample]).progress(0).build()
-        Project project2 = new Project.Builder("Q1244","This is a test project 2",[sample]).progress(1).build()
-        Project project3 = new Project.Builder("Q1233","This is a test project 3",[sample]).progress(0.9).failedSamples(1).build()
+    private void fetchProjectData() {
+        projects.clear()
+        projects.addAll(projectResourceService.iterator())
```

Comment:
These steps should also happen in the respective subscription to the service. E.g.
```
projectResourceService.subscribe({ projects.add($it) }, Topic.PROJECT_ADDED)
```